src/test/java/io/r2dbc/postgresql/codec/EnumCodecUnitTests.java

```diff
@@ -40,6 +52,70 @@ void shouldRejectMultipleMappingForTypeName() {
         Assertions.assertThatIllegalArgumentException().isThrownBy(() -> builder.withEnum("foo", MyOtherEnum.class));
     }
 
+    @Test
+    void shouldRegisterCodecAsFirst() {
+        CodecRegistrar codecRegistrar = EnumCodec
+            .builder()
+            .withEnum("foo", MyEnum.class, RegistrationPriority.FIRST)
+            .build();
+
+        ByteBufAllocator mockByteBufAllocator = Mockito.mock(ByteBufAllocator.class);
+        CodecRegistry mockCodecRegistry = Mockito.mock(CodecRegistry.class);
+
+        MockPostgresqlStatement mockPostgresqlStatement = MockPostgresqlStatement.builder()
+            .result(MockPostgresqlResult.builder()
+                .rowMetadata(MockRowMetadata.empty())
+                .row(MockRow.builder()
+                    .identified("oid", Integer.class, 42)
+                    .identified("typname", String.class, "foo")
+                    .identified("typcategory", String.class, "E")
+                    .build())
+                .build())
+            .build();
+        MockPostgresqlConnection mockPostgresqlConnection = new MockPostgresqlConnection(mockPostgresqlStatement);
+
+        Publisher<Void> register = codecRegistrar.register(mockPostgresqlConnection, mockByteBufAllocator, mockCodecRegistry);
+        StepVerifier.create(register).verifyComplete();
+
+        Mockito.verify(mockCodecRegistry, Mockito.only()).addFirst(Mockito.any(EnumCodec.class));
```

Comment:
Care to switch to static imports for Mockito?

Reply:
done